docs/proposals/20190801-BootSequence.md

```diff
@@ -0,0 +1,113 @@
+# Proposal: Boot Sequence
+
+Author: Andrew Rynhard
+
+## Abstract
+
+Ensuring proper order of tasks in boot and shutdown events is currently a problem.
+This document proposes a boot sequence management system that can be used to boot and shutdown a node in the proper order.
+
+## Background
+
+The main routine of `machined` was very script-like in the sense that it was a very long sequence of conditionals and functions.
+Lacking any kind of abstraction, a lot of code was repeated, conditionals were all over the place and hard to make sense of.
+This poses problems when we start to look at what it would take to bring a node back to its original state of having zero mounts, networking, processes (except PID 1 of course), etc.
+Undoing what is done only exacerbates all the previously mentioned problems.
+
+## Proposal
+
+This proposal builds on the notion of phases, tasks, and services by outlining yet another abstraction that ties these together.
+I propopse an `struct` aptly named `BootSequencer`.
+The definition might look something like the following:
+
+```go
+const (
+    MaxLevel = 5
+)
+
+type Status int
+
+const (
+    Pass Status = iota
+    Fail
+)
+
+type Level struct {
+    status Status
+    phases []phase.Phase
```

Comment:
should it be `phases [][]phase.Phase`?

Reply:
I think so ... building out an example go file and will be sure it compiles. I will copy over here when done.